### DISCUSSION TITLE
New seed software

### DISCUSSION THREAD

**Discussion by philipzae** [Xubuntu/xubuntu-default-settings#22] [Ideas]
Decide on new software that should be included with the default installation.

**Comment by philipzae** [comment]
[Proposed in ML](https://lists.ubuntu.com/archives/xubuntu-devel/2020-March/011877.html)
Disk Management (gnome-disk-utility)
Space Usage (baobab or mate-disk-usage-analyzer)
GUI Package Manager (synaptic)
Clipman Plugin (xfce4-clipman-plugin)
Global Menu Plugin (xfce4-appmenu-plugin)

**Proposed on IRC** [[log 1](https://irclogs.ubuntu.com/2020/03/10/%23xubuntu-devel.txt), [log 2](https://irclogs.ubuntu.com/2020/03/22/%23xubuntu-devel.txt)]
Adwaita theme (adwaita-icon-theme-full) [[bug](https://bugs.launchpad.net/ubuntu/+source/xubuntu-meta/+bug/1866814)]

**Additional Considerations**
Neofetch - Main system info app used in screenshots and will resolve the appearance of Xubuntu's logo [[bug](https://github.com/dylanaraps/neofetch/issues/1461)]

**Games**
- Consider removing Mines and Sudoku as they are included in SGT Collection
- Addition of [Gnome Games](https://wiki.gnome.org/Apps) like Aisleriot, Chess (gnome-chess), Quadrapassel, Nibbles (gnome-nibbles), and Four in a Row (four-in-a-row)

**User Requests**
Redshift (Monitor Color Temperature) - available in Xfce distros like Linux Mint and Amarok Linux
Timeshift (System Restore) - available in Xfce distros like Linux Mint and Amarok Linux

**Default Apps Research**
https://docs.google.com/spreadsheets/d/1TiVcXNSfiDkwIIonz5WyBFAcTkg-hw7iIVzDH6_gsME/edit#gid=0

**Comment by philipzae** [comment]
Wouldn't be useful to ship mate's disk usage app (mate-disk-usage-analyzer) as it isn't a standalone package and comes in mate-utils, which also has mate's dictionary, search tool, system log, and screenshot app. But if we want to also ship mate's system log app, then it may make sense.

**Comment by bluesabre** [comment]
I did some space research for each of the proposals for a bit more background here:

- gnome-disk-utility: 1,999 kB
- baobab: 737 kB
- synaptic: 6,346 kB
- xfce4-clipman-plugin: 1,018 kB
- xfce4-appmenu-plugin: 2,201 kB
- adwaita-icon-theme-full: 6,936 kB
- neofetch: 5,307 kB
- Removing gnome-mines and gnome-sudoku: +1,745 kB
- aisleriot: 54.6 MB
- gnome-chess: 1,538 kB
- quadrapassel: 885 kB
- gnome-nibbles: 1,456 kB
- four-in-a-row: 487 kB
- redshift-gtk: 1,882 kB
- timeshift: 3,323 kB

A grand total of 90.2 MB additional installed space usage with all changes included.

And then some space metrics on our current (no changes made above) installation:
- ISO size: 1.55 GiB
- Fully installed and updated disk usage: 7.8 GiB

**Comment by philipzae** [comment]
Thanks for commenting Sean, though the list mentioned in the second comment isn't the full list, as I hadn't done the research in the spreadsheet as yet, so please do have a look over it.

I'm pulling in my the recommendations from the spreadsheet as well as those in the [UI/UX google doc](https://docs.google.com/document/d/1hzvQURi8l--GrfqsxP9c0tWNUzrkbZjZTiVxGUegALg/edit#) and have started put them into this spreadsheet for easy viewing.
https://docs.google.com/spreadsheets/d/145vm29WnxgYm4Y_8guaGDAblZT5sRjZ3jhKRktgX1No/edit#gid=0

The size of neoftetch is 69.3kb if you don't pull in the recommended chafa, or is it mandatory to pull in recommended and not only depends.

**Comment by ghost** [comment]
GUI Package Manager (synaptic)
Clipman Plugin (xfce4-clipman-plugin)
Exaile
No games

**Comment by philipzae** [comment]
@ghost thanks for the input. synaptic and clipman are already in the list of additions. What is Exaile as I didnt find it in the repository.

**Comment by bluesabre** [comment]
> @ghost thanks for the input. synaptic and clipman are already in the list of additions. What is Exaile as I didnt find it in the repository.

[Exaile](https://www.exaile.org/) is a music player / media library that used to ship with Xubuntu. It went unmaintained for years and was dropped from the [Debian repositories](https://tracker.debian.org/pkg/exaile) in 2016.

**Comment by bluesabre** [comment]
> Thanks for commenting Sean, though the list mentioned in the second comment isn't the full list, as I hadn't done the research in the spreadsheet as yet, so please do have a look over it.
> 
> I'm pulling in my the recommendations from the spreadsheet as well as those in the [UI/UX google doc](https://docs.google.com/document/d/1hzvQURi8l--GrfqsxP9c0tWNUzrkbZjZTiVxGUegALg/edit#) and have started put them into this spreadsheet for easy viewing.
> https://docs.google.com/spreadsheets/d/145vm29WnxgYm4Y_8guaGDAblZT5sRjZ3jhKRktgX1No/edit#gid=0
> 
> The size of neoftetch is 69.3kb if you don't pull in the recommended chafa, or is it mandatory to pull in recommended and not only depends.

A few comments from the spreadsheet.

- First off, great research here. This is really well done and informative.
- We don't ship any QT components in Xubuntu since they bring along quite a few other dependencies.
- Instead of qt5ct, we'd be more likely to ship qt5-style-plugins which automatically adapts the active GTK theme for QT apps.
- Instead of shipping the Telegram desktop client, we could include a bookmark in our Firefox defaults to a Xubuntu channel if we move forward with it.
- Are there any configuration requirements for pm-utils, powertop, and tlp?

**Comment by philipzae** [comment]
As mentioned in our zoom discussion, lets see if there are still opposition to bringing in the Qt libs, like the same ones used by both qt5ct and qt5-style-plugins. They take up 45mb of space.

**Comment by shawnfromnh1** [comment]
My rules on my system are no extra software I will not use. I usually remove any office or chat apps immediately, I only use SMPlayer which is not a huge program and I use that for audio and video since most video apps are just overdone with options I find no use for, click a song and it plays is my only need no album covers or band info just play the song or directory which SMPlayer does both well. Now games the only solitaire I ever use is PysolFC it has auto drop option and you can disable sound and removal of splash screens and I can adjust the speed of the game to the point of no animation even and the right click to move a card to the piles is great since I only use it when I see an actual move and it saves a lot of strain on repetitive movement of wrists and the arm moving cards manually and Astro Menace is another game I often install though most of the games listed I would remove right after install is finished I don't like steam added to a distro, it's updates are large and if I'm not using it then it's sometimes several hundred megabytes. As for image viewers Gthumbs is my only one I like personally, and I have gimp, blender, Krita, and Inkscape but the average person would not want or need them so not recommending, as for system Gparted a must and gnome disk utility is a good idea since it's a well made functional program, as well as synaptic I always add. Now multimedia I have a few but I'll just suggest Handbrake and Piviti video editor with lots of filters but a bit of a learning curve but not excessively bad for someone that has done editing previously. Another add would be for me Audacity music editor small program and very useful. Just me but everything but clipboard manager I do not use at all though I do install Nautilus since XFCE has for searching and catfish does seem to work efficiently and Nautilus is a good file manager and has an exceptionally fast search function that does what it should do including sub-directories in the root system area. I find most "not a KDE fan" XFCE distro's lacking significantly on themes and icons so I would suggest a bunch of more normal looking icon themes as in the icons can easily be connected with the app they are there for not some strange icon for Firefox like a few I've used had on an unusual Thunar icon and themes a selection of themes with red blue green and other colors for variety would set the distro apart from other more sparse XFCE distro's. I did not see it listed but XFCE Goodies is a great package and adds a lot of functional stuff to XFCE that round out the DE to make it more usable and fun to use with ease. Another thing that is great is Orage panel clock since it has options for the text size and color which is good for me over 50 to see at a glance 28 pt font in a color that stands out since small clocks I have to focus since my eyesight is weaker now and the regular clock in the items is almost option less for me nothing in it is useful with text that is smaller than I need. I know it's long but I guess it's do it right and commit to helping on suggestions or just why make one suggestion when this is for data on users and what they like or not and it's great you are asking this since it's not wasting bandwidth with stuff no one uses or might not want till the need occurs making the iso smaller but also have that extra stuff that makes the distro more likely to be recommended to another person that is looking for something with just the stuff someone would need and I find office apps and chat are the most not wanted stuff when talking to others on linux distros and are likely a waste to have them on the ISO since they both can easily be downloaded after installation with synaptic so only the ones that need that large office suite are using the bandwidth to get it and not everyone plus if they update first then remove those chat and office apps that is more server bandwidth wasted.

**Comment by philipzae** [comment]
@shawnfromnh1 thanks for the feedback. I'm an SMPlayer user as well. I'm also a gThumb user as well, but use the forked version by Linux Mint called Pix. ;D

I use audacity, handbrake and kdenlive, but know that most users aren't into audio/video editing and conversion, but we should definitely highlight such multimedia apps that we do recommend documentation, etc. Audio and video conversion was more highly used in the old days when devices had limited playback functionality, but that's not the case nowadays with mp4 and mp3 being the dominant media formats that are playable everywhere.

More UI and icon themes are definitely wanted. Do you have any suggestions?

Most of xfce4-goodies dependent packages are already shipped by default, so if any of the ones below you think should be shipped by default, do let me know.
```
xfce4-cpufreq-plugin
xfce4-datetime-plugin
xfce4-diskperf-plugin
xfce4-fsguard-plugin
xfce4-goodies
xfce4-sensors-plugin
xfce4-smartbookmark-plugin
xfce4-timer-plugin
xfce4-wavelan-plugin
```

When it comes to the office apps, being able to open a text document or spreadsheet that you may have downloaded from the internet is a valid reason for shipping it by default. Another reason would be that xubuntu is being installed on school computers which would need to be equipped with an office suite for students to work with. Another reason would be that other OSes, e.g. Mac OS, ship with a pre-installed office suite. I am primarily a google docs users both personally and professionally, but there is always times when i end up opening a document in libreoffice, so having it there pre-installed is always useful. Also comes in handy when i'm offline and want to create or edit a document. Xubuntu does come with a minimal install version called [Xubuntu Core](https://unit193.net/xubuntu/) which those who prefer a smaller ISO size and only want to basics to build up from can do so with.

**Comment by philipzae** [comment]
@skellat has suggested we ship printer-driver-gutenprint, which is shipped with Lubuntu and Ubuntu MATE.

Its 1500kb download and 9988kb space.

**Comment by philipzae** [comment]
Seed [patch](https://bugs.launchpad.net/ubuntu/+source/xubuntu-meta/+bug/1914424) for themes.
https://github.com/philipzae/xubuntu-seed/commit/259acb27c7d4f4e22dbf93603726860575a2a9e3

**Comment by philipzae** [comment]
Seed [patch](https://bugs.launchpad.net/ubuntu/+source/xubuntu-meta/+bug/1916991) for games.
https://github.com/philipzae/xubuntu-seed/commit/a7ff1df865c0a6785a54c28a1e46b2e50a2c2f16

**Comment by bluesabre** [comment]
@philipzae I converted this issue to a discussion since it's a better fit. From your proposals so far, what's left? Let's discuss from here.

**Reply by philipzae** [reply]
Everything is found in the [spreadsheet](https://docs.google.com/spreadsheets/d/145vm29WnxgYm4Y_8guaGDAblZT5sRjZ3jhKRktgX1No/edit#gid=0) and awaits finalizing the particular app from each category that is being suggested. The games, internet and theming sections have pending patches that are awaiting approval.

**Reply by bluesabre** [reply]
Vote requested for the games from @Xubuntu/xubuntu-team here: https://lists.ubuntu.com/archives/xubuntu-devel/2021-July/012093.html
